sourced/ml/transformers/basic.py

```diff
@@ -161,10 +161,15 @@ def __init__(self, fields: Union[list, tuple], **kwargs):
         super().__init__(**kwargs)
         self.fields = fields
 
-    def __call__(self, df: DataFrame) -> DataFrame:
-        res = df.select(self.fields)
+    def __call__(self, rdd: RDD) -> RDD:
+        def select_fields(row):
+            res_row = {}
+            for field in self.fields:
+                res_row[field] = getattr(row, field)
```

Comment:
Since the tests failed and you have to change the code, how about turning this into a dict comprehension?

```python
return Row(**{f: getattr(row, f) for f in self.fields})
```